### ISSUE TITLE
nginx 1.9.x support

### ISSUE THREAD

**Issue by gfrankliu** [weibocom/nginx-upsync-module#53]
There are many nginx-upsync-module changes in master since the nginx-1.9.x branch was created early Feb. Is it possible to wrap those nginx-1.9.x changes with ifdef's so that we can maintain one code base?

**Comment by xiaokai-wang** [comment]
There will be a new branch for nginx-1.9.x and the old one will be merged into the new one, later.

Thanks.

**Comment by xiaokai-wang** [comment]
There will be a new branch for nginx-1.9.x and the old one will be merged into the new one, later.

Thanks.

**Comment by xiaokai-wang** [comment]
There will be a new branch for nginx-1.9.x and the old one will be merged into the new one, later.

Thanks.

**Comment by xiaokai-wang** [comment]
There will be a new branch for nginx-1.9.x and the old one will be merged into the new one, later.

Thanks.

**Comment by xiaokai-wang** [comment]
There will be a new branch for nginx-1.9.x and the old one will be merged into the new one, later.

Thanks.

**Comment by xiaokai-wang** [comment]
There will be a new branch for nginx-1.9.x and the old one will be merged into the new one, later.

Thanks.

**Comment by xiaokai-wang** [comment]
There will be a new branch for nginx-1.9.x and the old one will be merged into the new one, later.

Thanks.

**Comment by xiaokai-wang** [comment]
There will be a new branch for nginx-1.9.x and the old one will be merged into the new one, later.

Thanks.

**Comment by xiaokai-wang** [comment]
There will be a new branch for nginx-1.9.x and the old one will be merged into the new one, later.

Thanks.

**Comment by xiaokai-wang** [comment]
There will be a new branch for nginx-1.9.x and the old one will be merged into the new one, later.

Thanks.

**Comment by xiaokai-wang** [comment]
There will be a new branch for nginx-1.9.x and the old one will be merged into the new one, later.

Thanks.

**Comment by xiaokai-wang** [comment]
There will be a new branch for nginx-1.9.x and the old one will be merged into the new one, later.

Thanks.

**Comment by xiaokai-wang** [comment]
There will be a new branch for nginx-1.9.x and the old one will be merged into the new one, later.

Thanks.

**Comment by xiaokai-wang** [comment]
There will be a new branch for nginx-1.9.x and the old one will be merged into the new one, later.

Thanks.

**Comment by xiaokai-wang** [comment]
There will be a new branch for nginx-1.9.x and the old one will be merged into the new one, later.

Thanks.

**Comment by xiaokai-wang** [comment]
There will be a new branch for nginx-1.9.x and the old one will be merged into the new one, later.

Thanks.

**Comment by xiaokai-wang** [comment]
There will be a new branch for nginx-1.9.x and the old one will be merged into the new one, later.

Thanks.

**Comment by xiaokai-wang** [comment]
There will be a new branch for nginx-1.9.x and the old one will be merged into the new one, later.

Thanks.

**Comment by xiaokai-wang** [comment]
There will be a new branch for nginx-1.9.x and the old one will be merged into the new one, later.

Thanks.

**Comment by xiaokai-wang** [comment]
There will be a new branch for nginx-1.9.x and the old one will be merged into the new one, later.

Thanks.

**Comment by xiaokai-wang** [comment]
There will be a new branch for nginx-1.9.x and the old one will be merged into the new one, later.

Thanks.

**Comment by xiaokai-wang** [comment]
There will be a new branch for nginx-1.9.x and the old one will be merged into the new one, later.

Thanks.

**Comment by xiaokai-wang** [comment]
There will be a new branch for nginx-1.9.x and the old one will be merged into the new one, later.

Thanks.

**Comment by xiaokai-wang** [comment]
There will be a new branch for nginx-1.9.x and the old one will be merged into the new one, later.

Thanks.

**Comment by xiaokai-wang** [comment]
There will be a new branch for nginx-1.9.x and the old one will be merged into the new one, later.

Thanks.

**Comment by xiaokai-wang** [comment]
There will be a new branch for nginx-1.9.x and the old one will be merged into the new one, later.

Thanks.

**Comment by xiaokai-wang** [comment]
There will be a new branch for nginx-1.9.x and the old one will be merged into the new one, later.

Thanks.

**Comment by xiaokai-wang** [comment]
There will be a new branch for nginx-1.9.x and the old one will be merged into the new one, later.

Thanks.

**Comment by xiaokai-wang** [comment]
There will be a new branch for nginx-1.9.x and the old one will be merged into the new one, later.

Thanks.

**Comment by xiaokai-wang** [comment]
There will be a new branch for nginx-1.9.x and the old one will be merged into the new one, later.

Thanks.

**Comment by xiaokai-wang** [comment]
There will be a new branch for nginx-1.9.x and the old one will be merged into the new one, later.

Thanks.

**Comment by gfrankliu** [comment]
The current 1.9.x branch seems to at least work alone with nginx 1.9, and the health check module https://github.com/xiaokai-wang/nginx_upstream_check_module alone also works with nginx 1.9. But if I both modules with nginx 1.9, nginx keeps crashing.
```
2016/04/08 06:05:18 [alert] 12080#0: worker process 12690 exited on signal 11
2016/04/08 06:05:18 [alert] 12080#0: worker process 12691 exited on signal 11
2016/04/08 06:05:18 [alert] 12080#0: worker process 12692 exited on signal 11
2016/04/08 06:05:18 [alert] 12080#0: worker process 12693 exited on signal 11
2016/04/08 06:05:18 [alert] 12080#0: worker process 12694 exited on signal 11
2016/04/08 06:05:18 [alert] 12080#0: worker process 12695 exited on signal 11
2016/04/08 06:05:18 [alert] 12080#0: worker process 12696 exited on signal 11
2016/04/08 06:05:18 [alert] 12080#0: worker process 12697 exited on signal 11
2016/04/08 06:05:18 [alert] 12080#0: worker process 12698 exited on signal 11
2016/04/08 06:05:18 [alert] 12080#0: worker process 12699 exited on signal 11
```

**Comment by xiaokai-wang** [comment]
Ok, got, thanks.

**Comment by gfrankliu** [comment]
The new nginx-upsync-1.9.x branch seem to work. Thanks!

BTW, nginx finishes the 1.9 dev cycle today and released stable 1.10.0 (it is basically 1.9.15). Maybe nginx-upsync-module can branch out 1.8 to track nginx-1.8 changes, and merge 1.9 stuff back to master so that master branch supports latest nginx 1.10 stable?

**Comment by xiaokai-wang** [comment]
Yes, just like stealing my idea.
